stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -565,13 +565,25 @@ function repl_eval_ex(@nospecialize(ex), context_module::Module)
     interp = REPLInterpreter(result)
     frame = CC.InferenceState(result, src, #=cache=#:no, interp)::CC.InferenceState
 
-    CC.typeinf(interp, frame)
+    # NOTE Use the fixed world here to make `REPLInterpreter` robust against
+    #      potential invalidations of `Core.Compiler` methods.
+    Base.invoke_in_world(COMPLETION_WORLD[], CC.typeinf, interp, frame)
 
     result = frame.result.result
     result === Union{} && return nothing # for whatever reason, callers expect this as the Bottom and/or Top type instead
     return result
 end
 
+# `COMPLETION_WORLD[]` will be initialized within `__init__`
+# (to allow us to potentially remove REPL from the sysimage in the future).
```

Comment:
```suggestion
# `COMPLETION_WORLD[]` will be initialized within `__init__`
# (to allow us to potentially remove REPL from the sysimage in the future).
# Note that inference from the `code_typed` call below will use the current world age
# rather than `typemax(UInt)`, since `Base.invoke_in_world` uses the current world age
# when the given world age is higher than the current one.
```